src/main/java/backend/controller/PostingController.kt

```diff
@@ -67,20 +67,26 @@ class PostingController(private val postingService: PostingService,
         val posting = postingService.createPosting(user, body.text, body.uploadMediaTypes, locationCoord, clientDate)
         posting.media.forEach { it.uploadToken = getSignedJwtToken(JWT_SECRET, it.id.toString()) }
 
-        return PostingView(posting, null)
+        return PostingView(posting, null, user.account.id)
     }
 
     /**
      * GET /posting/{id}/
      * Gets posting by id
      */
     @GetMapping("/{id}/")
-    fun getPosting(@PathVariable("id") id: Long, @RequestParam(value = "userid", required = false) userId: Long?): PostingResponseView {
+    fun getPosting(@PathVariable("id") id: Long,
+                   @RequestParam(value = "userid", required = false) userId: Long?): PostingResponseView {
```

Comment:
This should not be userid as request param, but can be fetched from spring context of the currently authenticated user.

Maybe try out @AuthenticationPrincipal and see what happens